### ISSUE TITLE
Sorbet wrongly flags Array.new with array as input

### ISSUE THREAD

**Issue by ivoanjo** [sorbet/sorbet#1913]
#### Input

[→ View on sorbet.run](https://sorbet.run/#Array.new(%5B%22hello%22%2C%20%22World%22%5D)%0A)

```ruby
Array.new(["hello", "World"])
```

#### Observed output

```ruby
editor.rb:1: Expected Integer but found [String("hello"), String("World")] for argument arg0 https://srb.help/7002
     1 |Array.new(["hello", "World"])
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    https://github.com/sorbet/sorbet/tree/master/rbi/core/array.rbi#L1248: Method Array#initialize (overload.1) has specified arg0 as Integer
    1248 |        arg0: Integer,
                  ^^^^
  Got [String("hello"), String("World")] originating from:
    editor.rb:1:
     1 |Array.new(["hello", "World"])
                  ^^^^^^^^^^^^^^^^^^
Errors: 1
```

#### Expected behavior

As documented in <https://ruby-doc.org/core-2.5.6/Array.html#method-c-new>, it's possible to call the `Array.new` method with an array, in addition to the more common `size` argument.

**Comment by jez** [comment]
https://sorbet.org/docs/faq#it-looks-like-sorbet-s-types-for-the-stdlib-are-wrong

**Comment by marianosimone** [comment]
This was fixed in #3223
